test/Main.hs

```diff
@@ -3,11 +3,20 @@ module Main where
 import Test.QuickCheck
 import Data.Aeson
 import Nix.Diff.Types
+import Nix.Diff.Transformations
 
 main :: IO ()
-main = quickCheck prop_DecodeEncodeID
+main = do
+  quickCheck prop_DecodeEncodeID
+  quickCheck prop_foldingComesToFixedPoint
 
 prop_DecodeEncodeID :: DerivationDiff -> Bool
 prop_DecodeEncodeID dd = case decode (encode dd) of
   Nothing -> False
   Just dd' -> dd == dd'
+
+prop_foldingComesToFixedPoint :: DerivationDiff -> Bool
+prop_foldingComesToFixedPoint dd = 
+   (op dd) == (op . op $ dd)
+  where
+    op = foldAlreadyComparedSubTrees
```

Comment:
A small nitpick: newline at the end of file is missing. Same for `src/Nix/Diff/Transformations.hs`